### ISSUE TITLE
UPDATE_FAILED error: api [unauthRoleName, authRoleName] do not exist in the template

### ISSUE THREAD

**Issue by matt-d-webb** [aws-amplify/amplify-cli#7915]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

yarn

### If applicable, what version of Node.js are you using?

v14.16.0

### Amplify CLI Version

5.3.0

### What operating system are you using?

Mac

### Amplify Categories

api

### Amplify Commands

push

### Describe the bug

Trying to add a new API via `amplify add api`, had done a backend sync prior with `amplify pull`. Get the following error in the console:

`Parameters: [authRoleName, unauthRoleName] do not exist in the template`

Attempted to follow suggestions in related issue: https://github.com/aws-amplify/amplify-cli/issues/2519

The `parameters.json` file present with json:

```
{
    "authRoleName": {
        "Ref": "AuthRoleName"
    },
    "unauthRoleName": {
        "Ref": "UnauthRoleName"
    }
}

```

Tried:
1. amplify update auth - issue on push `Resource is not in the state stackUpdateComplete`
2. amplify remove api + amplify push - Successful but repeat steps to re add api results in above issue reported

### Expected behavior

Amplify push deploys successfully.

### Reproduction steps

1. amplify add api
2. add to existing api
2. amplify push

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by cjihrig** [comment]
If this is a REST API, please try removing those two entries from the API's `parameters.json` file.

**Comment by matt-d-webb** [comment]
@cjihrig I get the following error:

`Parameters: [authRoleName, unauthRoleName] must have values`

**Comment by cjihrig** [comment]
@matt-d-webb in your original post you mentioned this error:

> Parameters: [authRoleName, unauthRoleName] do not exist in the template

Did this error come from the REST API, or some other resource?

**Comment by matt-d-webb** [comment]
@cjihrig this came from the REST API. I have tried to revert the changes which has resulted in a different error.

> Parameters: [breakCircularDependency, aliasAttributes] must have values

![image](https://user-images.githubusercontent.com/36933715/129101497-9345c0dc-e5a6-4570-860d-104fcf604cd0.png)

I figure this relates to attempts to fix this following suggestions here #2519 and doing `amplify update auth`

**Comment by akshbhu** [comment]
Hi @matt-d-webb 

If you have deployed any previous changes , you can use  ```amplify env pull ``` command to get your project back to the previously deployed state. For more information you can check [here](https://docs.amplify.aws/cli/teams/commands#pull-the-environment-from-the-cloud):
